server.py

```diff
@@ -0,0 +1,186 @@
+
+def send_data_ws(ws, data):
+    data_json = json.dumps(data)
+    ws.send(data_json)
+
+class Client:
+    """Client class for users that connect to devices"""
+
+    client_id = None
+    device = None
+    ws = None
+
+    def __init__(self, ws):
+        self.ws = ws
+
+    def send_device_info(self):
+        client_ws = self.ws
+        device_info_msg = {}
+        device_info_msg['message_type'] = 'device_info'
+        device_info_msg['device_info'] = self.device.device_info
+        send_data_ws(client_ws, device_info_msg)
+    
+    def forward_client_message(self, message):
+        device_ws = self.device.ws
+        client_msg = {}
+        client_msg['message_type'] = 'client_msg'
+        client_msg['client_id'] = self.client_id
+        client_msg['payload'] = message['payload']
+        send_data_ws(device_ws, client_msg)
+
+    def handle_connect(self, message):
+        """Connects client user to device"""
+        
+        if 'device_id' not in message:
+            send_error('Missing device_id field.')
+            return
+
+        device_id = message['device_id']        
+        if self.client_id:
+            send_error('Attempt to connect to multiple devices over same websocket.')
+        else:
+            send_server_config(self.ws)
+            if device_id not in devices:
+                send_error(f'Device id {device_id} not registered.')
+            else:
+                self.device = devices[device_id]
+                self.device.register_client(self)
+                print(f'Connected client {self.client_id} to device {self.device.device_id}.')
+                self.send_device_info()
+
+    def handle_forward(self, message):
+        """Handle forward for client"""
+
+        if not self.client_id:
+            send_error('No device associated to client.')
+        elif 'payload' not in message:
+            send_error('Missing payload field.')
+        else:
+            self.forward_client_message(message)
+            print(f'Forwarded message from client {self.client_id} to device {self.device.device_id}.')
+
+    def process_request(self, message):
+        if 'message_type' not in message:
+            send_error('Missing field message_type')
+        elif message['message_type'] == 'connect':
+            self.handle_connect(message)
+        elif message['message_type'] == 'forward':
+            self.handle_forward(message)
+        else:
+            send_error(f'Unknown message type: {message["message_type"]}')
+
+
+class Device:
+
+    client_number = 1
+    device_id = None
+    device_info = None
+    clients = {}
+
+    def __init__(self, ws):
+        self.ws = ws
+
+    def forward_device_message(self, message):
+        client_id = message['client_id']
+        client_ws = self.clients[client_id].ws
+        device_msg = {}
+        device_msg['message_type'] = 'device_msg'
+        device_msg['payload'] = message['payload']
+        send_data_ws(client_ws, device_msg)
```

Comment:
nit: same as above, the client class should be the only one accessing the client socket